keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -320,28 +344,54 @@ class ApplicationService(
     version: String,
     environments: Set<ArtifactSummaryInEnvironment>
   ): ArtifactVersionSummary {
-    val artifactSupplier = artifactSuppliers.supporting(artifact.type)
-    val releaseStatus = repository.getReleaseStatus(artifact, version)
-    val artifactInstance = repository.getArtifactInstance(artifact.name, artifact.type, version, releaseStatus)
-      ?: throw InvalidSystemStateException("Loading artifact version $version failed for known artifact $artifact.")
 
+    val artifactSupplier = artifactSuppliers.supporting(artifact.type)
+    val artifactInstance = getArtifactInstance(artifact, version)
+            ?: throw InvalidSystemStateException("Loading artifact version $version failed for known artifact $artifact.")
     return ArtifactVersionSummary(
-      version = version,
-      environments = environments,
-      displayName = artifactSupplier.getVersionDisplayName(artifactInstance),
-      createdAt = artifactInstance.createdAt,
-
-      // first attempt to use the artifact metadata fetched from the DB, then fallback to the default if not found
-      build = artifactInstance.buildMetadata
-        ?: artifactSupplier.parseDefaultBuildMetadata(artifactInstance, artifact.versioningStrategy),
-      git = artifactInstance.gitMetadata
-        ?: artifactSupplier.parseDefaultGitMetadata(artifactInstance, artifact.versioningStrategy)
-    )
-  }
+        version = version,
+        environments = environments,
+        displayName = artifactSupplier.getVersionDisplayName(artifactInstance),
+        createdAt = artifactInstance.createdAt,
+
+        // first attempt to use the artifact metadata fetched from the DB, then fallback to the default if not found
+        build = artifactInstance.buildMetadata
+          ?: artifactSupplier.parseDefaultBuildMetadata(artifactInstance, artifact.versioningStrategy),
+        git = artifactInstance.gitMetadata
+          ?: artifactSupplier.parseDefaultGitMetadata(artifactInstance, artifact.versioningStrategy)
+      )
+    }
 
   fun getApplicationEventHistory(application: String, limit: Int) =
     repository.applicationEventHistory(application, limit)
 
   private fun ConstraintState.toConstraintSummary() =
     StatefulConstraintSummary(type, status, createdAt, judgedBy, judgedAt, comment, attributes)
+
+  private fun getArtifactInstance(
+    artifact: DeliveryArtifact,
+    version: String
+  ): PublishedArtifact? {
+    val releaseStatus = repository.getReleaseStatus(artifact, version)
+    return repository.getArtifactInstance(artifact.name, artifact.type, version, releaseStatus)
+  }
+
+  // Generating a stash diff link between source and target versions (the order does matter!)
```

Comment:
Consider removing specific mentions of Stash in this code, since it must work regardless of the SCM being used.

Reply:
every SCM has a different comparison implementation. I didn't check how it was implemented in other SCM (like git etc.) as I think it's out of scope at the moment. I can address it in a follow-up PRs, if the team feels it's nessacery.

Reply:
Regardless of whether you address the specifics of other SCM systems later, my point was about encapsulation and extensibility. Why hard-code comments and implementations when you can easily make it generic and future-proof?

Reply:
@luispollo there's a cost to making something "generic and future-proof", and I think a lot of times it's hard to know what the future will look like. A generic implementation that is a little bit wrong is often pretty hard to debug and to change. On the other side, it's fairly easy to refactor to a generic implementation later on when it's needed.